core/plugins/org.polarsys.capella.core.platform.sirius.sirius.validation/src/org/polarsys/capella/core/platform/sirius/sirius/validation/ddiagram/ImagePathResolver.java

```diff
@@ -0,0 +1,122 @@
+      return;
+    EObject target = modelElements.get(0);
+    Optional<Session> sessionOpt = Session.of(target);
+
+    Diagnostic diagnostic = MarkerViewHelper.getDiagnostic(marker_p);
+    ImagePathWrappingStatus imagePathStatus = null;
+    if (diagnostic instanceof ConstraintStatusDiagnostic
+        && ((ConstraintStatusDiagnostic) diagnostic).getConstraintStatus() instanceof ImagePathWrappingStatus) {
+      imagePathStatus = (ImagePathWrappingStatus) ((ConstraintStatusDiagnostic) diagnostic).getConstraintStatus();
+    }
+
+    if (imagePathStatus != null && sessionOpt.isPresent()) {
+      TransactionalEditingDomain ted = sessionOpt.get().getTransactionalEditingDomain();
+
+      String notReachableImagePath = imagePathStatus.getNotReachableImagePath();
+      boolean fixSucceeded = FileProvider.getDefault().exists(new Path(notReachableImagePath));
+
+      if (!fixSucceeded) {
+        ImagePathTarget imagePathTarget = imagePathStatus.getImagePathTarget();
+        EAttribute eAttribute = imagePathStatus.getEAttribute();
+        if (ImagePathTarget.DREPRESENTATION_DESCRIPTOR.equals(imagePathTarget) && eAttribute != null
+            && target instanceof DRepresentationDescriptor) {
+
+          // open the editor and fix the image issue
+          DRepresentationDescriptor representationDescriptor = (DRepresentationDescriptor) target;
+          DialectUIManager.INSTANCE.openEditor(sessionOpt.get(), representationDescriptor.getRepresentation(),
+              new NullProgressMonitor());
+          fixSucceeded = ImageMarkerResolution.fixImagePathInRichText(representationDescriptor, eAttribute.getName(),
+              notReachableImagePath, ted);
+
+        } else if (ImagePathTarget.SEMANTIC_TARGET.equals(imagePathTarget) && eAttribute != null) {
+
+          // fix the image path for the semantic object
+          fixSucceeded = ImageMarkerResolution.fixImagePathInRichText(target, eAttribute.getName(),
+              notReachableImagePath, ted);
+
+        } else if (ImagePathTarget.WORKSPACE_IMAGE.equals(imagePathTarget) && target instanceof DSemanticDecorator) {
+
+          DRepresentation dRepresentation = getRepresentation(target);
+          if (dRepresentation != null) {
+            DialectUIManager.INSTANCE.openEditor(sessionOpt.get(), dRepresentation, new NullProgressMonitor());
+            fixSucceeded = ImageMarkerResolution.fixWorkspaceImagePath(target, notReachableImagePath, ted);
+          }
+        }
+      }
+
+      if (fixSucceeded) {
+        try {
+          marker_p.delete();
+        } catch (CoreException exception_p) {
+          _logger.error("Exception while deleting marker : " + exception_p.toString()); //$NON-NLS-1$
+        }
+      }
+    }
+  }
+
+  /**
+   * Browse the model upward (from the leaf to the root) and return the first representation found.
+   * 
+   * @return the representation if found, null otherwise.
+   */
+  private DRepresentation getRepresentation(EObject eObject) {
```

Comment:
can you replace EObject by DSemanticDecorator

Reply:
Done